feat(react-grid): improve column resizing (#2218)
DevExpress/devextreme-reactive: packages/dx-react-grid/docs/guides/column-resizing.md

> In [controlled mode](controlled-and-uncontrolled-modes.md), pass the column widths to the `TableColumnResizing` plugin's `columnWidths` property, and handle the `onColumnWidthsChange` event to control the column width state.

Reviewer: ```suggestion
In [controlled mode](controlled-and-uncontrolled-modes.md), pass the column widths to the `TableColumnResizing` plugin's `columnWidths` property and handle the `onColumnWidthsChange` event to manage the column width state.
```

.embedded-demo({ "path": "grid-column-resizing/controlled", "showThemeSelector": true })

## Resizing Modes

Columns can be resized in the following modes:

- `nextColumn`        
When a user resizes a column, the width of the next column changes.

- `widget`        

> When a user resizes a column, columns located on the right shift, but retain their widths. In this mode, column widths cannot be specified using the `auto` and percentage values.

Reviewer: ```suggestion
When a user resizes a column, columns on the right shift but retain their widths. In this mode, users cannot use `auto` and percentage values to specify column widths.
```

In [controlled mode](controlled-and-uncontrolled-modes.md), pass the column widths to the `TableColumnResizing` plugin's `columnWidths` property, and handle the `onColumnWidthsChange` event to control the column width state.

.embedded-demo({ "path": "grid-column-resizing/controlled", "showThemeSelector": true })

> ## Resize Modes

Reviewer 2: ```suggestion
## Resizing Modes
```

> ### Resizing Mode

Reviewer 3: I think we should extract this part from the `Basic` guide part. (`### Resizing Mode` -> `## Resize Modes``)

> `TableColumnResizing` plugin has two resizing mode, changing by `columnResizingMode` property. That property may be defined by two values: `widget` and `nextColumn`. In Widget mode when you resize column, all following columns shifts. Also in this mode, you can't use `auto` and `%` width size. In NextColumn mode when you resize column, you change size of next column. In this mode you can't change summary width of columns. The following demo shows booth resizing modes.

Reviewer 2: ```suggestion
Columns can be resized in the following modes:

- `nextColumn`        
When a user resizes a column, the width of the next column changes.

- `widget`        
When a user resizes a column, columns located on the right shift, but retain their widths. In this mode, column widths cannot be specified using the `auto` and percentage values.

To specify the resizing mode, use the `TableColumnResizing` plugin's `columnResizingMode` property. 

The following example demonstrates the resizing modes:
```
